### PR TITLE
📖 Readme

### PR DESCRIPTION
**Pull request by demariadaniel**

### CODE REVIEW

File: packages/iobio-react-components/README.md [hunk 1]
```diff
@@ -2,14 +2,52 @@
...
+
+- The `<IobioDataBroker />` component is necessary to analyze a given file. It receives an `alignmentUrl` string prop which should point to a BAM/CRAM file URL. The Iobio server performs an analysis on the given file, and the other Iobio components read this data.
+
+- Some visualization components (`histogram` and `percentBox`) require data keys that correspond to this analysis to know what data to read. These values can be found in `/utils/constants`
+
+- The Overture `Iobio React Components` library also adds basic Typescript definitions in iobio.d.ts. `iobio-charts` is not typed.
+
+### SSR Integrations
+
+- Integrations with Overture Stage revealed conflicts between web component usage in `iobio-charts` and Next.js server side rendering, which may be an issue in future projects. Iobio Charts uses DOM APIs that are not available when rendered on the server, and which create conflicts when rendered multiple times.
```
Comment (justincorrigible) [inline packages/iobio-react-components/README.md:32]:
```suggestion
- Integrations with Overture Stage revealed conflicts between web component usage in `iobio-charts` and Next.js server side rendering, which may be an issue in future projects. Iobio Charts uses a DOM API that is not available when rendered on the server, and which create conflicts when rendered multiple times.
```

File: packages/iobio-react-components/README.md [hunk 2]
```diff
@@ -2,14 +2,52 @@
...
+
+- These APIs include (but are not limited to `customElements`, `HTMLElement`, `Element` and `CssStyleSheet`.
```
Comment (justincorrigible) [inline packages/iobio-react-components/README.md:34]:
```suggestion
- This API is used by (but not limited to) `customElements`, `HTMLElement`, `Element` and `CssStyleSheet`.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Overture Iobio Components
+
+- A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by https://iobio.io/
```
Comment (justincorrigible) [inline README.md:3]:
```suggestion
- A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by [IOBIO](https://iobio.io/)
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+- This project is part of the [Overture](https://www.overture.bio/) research software ecosystem. See our [related products](#related-products) for more information on how Overture is helping organize data and enable discovery.
```
Comment (justincorrigible) [inline README.md:5]: what `#related-products` are you referring to? maybe the link is meant to go somewhere in overture.bio? or is it WIP in this readme?

**Review by justincorrigible — COMMENTED** [review]
couple of suggestions to help ease reading

File: README.md (hunk 3 above)
Comment (joneubank) [reply README.md:3]: Also, end the sentence with a period.

File: packages/iobio-react-components/README.md [hunk 5]
```diff
@@ -2,14 +2,52 @@
...
+`
+module.exports = withPlugins([withTranspileModules], {
+	webpack: (config, options) => {
+		if (options.isServer) {
+			const { JSDOM } = jsdom;
+			const dom = new JSDOM('', { url: 'http://localhost/' });
+      global.customElements = global.window.customElements;
+`
+
+- This solution works in Next 12.1 and may be updated in future iterations.
```
Comment (joneubank) [inline packages/iobio-react-components/README.md:50]: Markdown formatting seems wrong here. There are comments outside the code block, and the code block will need triple back ticks.

**Review by joneubank — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (demariadaniel) [reply README.md:5]: Sorry lazy copy from Stage. I'm going to update to use the template: https://main--overturedev.netlify.app/docs/standards/software/readme/

File: packages/iobio-react-components/README.md (hunk 5 above)
Comment (demariadaniel) [reply packages/iobio-react-components/README.md]: Fixed, thanks

File: packages/iobio-react-components/README.md [hunk 6]
```diff
@@ -2,14 +2,52 @@
 
 ## Setup + Local Development
 
+### Requirements:
+
+- PNPM 9+
+- Node 20+
+
 - Install dependencies: `pnpm i`
```
Comment (ciaranschutte) [inline packages/iobio-react-components/README.md:10]: should this be `ci`? or does pnpm figure out all that npm `ci` vs `i` stuff itself

File: packages/iobio-react-components/README.md [hunk 7]
```diff
@@ -2,14 +2,52 @@
...
 ## React Components
 
-- Stub for React Component documentation.
-<!-- TODO: Write Docs (include pnpm, monorepo, etc) -->
+- This package exports a set of React + TypeScript components to facilitate the integration of the Iobio Charts library of "web components" into your UI app.
+
+- The Demo App at `iobio-components/apps/bam.iobio` shows an example integration which closely mirrors the integration at bam.iobio.io
+
+- The `<IobioDataBroker />` component is necessary to analyze a given file. It receives an `alignmentUrl` string prop which should point to a BAM/CRAM file URL. The Iobio server performs an analysis on the given file, and the other Iobio components read this data.
+
```
Comment (ciaranschutte) [inline packages/iobio-react-components/README.md:25]: I think `..and the other Iobio components read this data` could be expanded on. It's not immediately clear or a super common that the React components are mostly wrapping web components for display, which are hooked in to events in the data broker components. If I came across this repo, without seeing it being developed, I would assume React (or React compatible libraries) were controlling the state.

File: packages/iobio-react-components/README.md [hunk 8]
```diff
@@ -2,14 +2,52 @@
...
+// next.config.js
+module.exports = withPlugins([withTranspileModules], {
+	webpack: (config, options) => {
+		if (options.isServer) {
+			const { JSDOM } = jsdom;
+			const dom = new JSDOM('', { url: 'http://localhost/' });
+      global.customElements = global.window.customElements;
+```
+
+- This solution works in Next 12.1 and may be updated in future iterations.
```
Comment (ciaranschutte) [inline packages/iobio-react-components/README.md:50]: I appreciate the detail in this and the further information for integration.
I do think it's a bit unfair to term these issues as "conflicts"

TLDR: let's just be explicit we only support client side rendering

Could we just be explicit and say these components only work client side? 
That's the extend of our work and what _we_ support with our lib.
Every framework has some form of SSR so I don't think these issues are unique to Next, we just don't support SSR.
And DOM apis are web browser not server apis.
This phrasing feels to me like we're pointing out bugs or issues. 
But the case is closer to "my gif won't play on this printed sheet of paper."

I think being **explicit that we only support client side rendering** should also cover integration for others with frameworks.

I don't feel great about recommending an entire DOM library dependency just for a frontend visualisation lib to work. It's also an extra dependency that isn't in the package.json of the package.

We're out of time now, but in the future maybe we could integrate it into our `iobio-component` lib because ,as I understand, right now it's a hard dependency to send `dom` event messages and "register" custom elements. It's also completely dependent on that library maintainer keeping up to speed with specs vs the browser companies. Take all of this with a grain of salt, I know we have limited time, just my 2cents.

File: packages/iobio-react-components/README.md (hunk 6 above)
Comment (demariadaniel) [reply packages/iobio-react-components/README.md:10]: PNPM is already managing that.

https://pnpm.io/cli/install
https://stackoverflow.com/questions/70154568/pnpm-equivalent-command-for-npm-ci

File: packages/iobio-react-components/README.md (hunk 8 above)
Comment (demariadaniel) [reply packages/iobio-react-components/README.md:50]: For context re: limited time, this is a break period which is ideal to identify issues for follow up when we resume dev work on this project.

Being explicit about 'client side only' is a very fair change to the readme.

This is included in the readme explicitly because it's an area for improvement and dev attention. I feel like there is a more optimal solution out there, but for use with Next 12 and our timeframes this worked. Ideally I'd update Stage to Next 14 to start.

I wanted to provide an example as a starting point. I don't think this is too out of line with other readmes I see on Github / NPM. I feel like guides for integration with different frameworks isn't unheard of.

It did cross my mind that the dom shim code in Stage next.config could possibly be shipped as an export from this library. So there's potential there to include 'jsdom' as a dependency and export the block from [Stage next.config](https://github.com/overture-stack/stage/blob/iobio/next.config.js#L19) as a helper function.

Sorry where are you referencing `DOM apis are web browser not server apis` ?

File: README.md (hunk 4 above)
Comment (demariadaniel) [reply README.md:5]: This is fixed, added an anchor to the related header

File: packages/iobio-react-components/README.md (hunk 7 above)
Comment (demariadaniel) [reply packages/iobio-react-components/README.md]: OK added some clarified language

File: packages/iobio-react-components/README.md (hunk 8 above)
Comment (demariadaniel) [reply packages/iobio-react-components/README.md]: I've rephrased this block but will review again.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+- A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by [IOBIO](https://iobio.io/).
+
+- This project is part of the [Overture](https://www.overture.bio/) research software ecosystem. See our [related products](#related-products) for more information on how Overture is helping organize data and enable discovery.
+
+</br>
+
+> 
+> <div>
+> <img align="left" src="ov-logo.png" height="90"/>
```
Comment (MitchellShiell) [inline README.md:11]:
```suggestion
> <img align="left" src="ov-logo.png" height="50"/>
```

Comment (MitchellShiell) [reply README.md:11]: This will keep the Overture Logo from interfering with the Repository structure title when viewing from the repo page

File: README.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+|[Song](https://github.com/overture-stack/song/)| Catalog and manage metadata associated to file data spread across cloud storage systems |
+|[Maestro](https://github.com/overture-stack/maestro/)| Organizing your distributed data into a centralized Elasticsearch index |
+|[Arranger](https://github.com/overture-stack/arranger/)| A search API with reusable search UI components |
+|[Stage](https://github.com/overture-stack/stage)| A React-based front-data portal UI |
+|[Lyric](https://github.com/overture-stack/lyric)| A data-agnostic tabular data submission system |
+|[Lectern](https://github.com/overture-stack/lectern)| A simple web browser UI that integrates Ego and Arranger |
+
+## Acknowledgements
+
+- Overture is supported by grant #U24CA253529 from the National Cancer Institute at the US National Institutes of Health.
```
Comment (MitchellShiell) [inline README.md:86]: Nitpick: bullet points typically reserved for unordered lists

```suggestion
Overture is supported by grant #U24CA253529 from the National Cancer Institute at the US National Institutes of Health.
```

**Comment by MitchellShiell** [comment]
Looks great happy to see the template in action, let me know if their's is anything you'd like to see that is not included, or should be removed, LGTM Dan!

File: README.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
+# Overture Iobio Components
+
+- A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by [IOBIO](https://iobio.io/).
+
+- This project is part of the [Overture](https://www.overture.bio/) research software ecosystem. See our [related products](#related-products) for more information on how Overture is helping organize data and enable discovery.
```
Comment (MitchellShiell) [inline README.md:5]:
```suggestion

```

File: README.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
+# Overture Iobio Components
+
+- A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by [IOBIO](https://iobio.io/).
```
Comment (MitchellShiell) [inline README.md:3]:
```suggestion
A set of reusable React components and command line tools for integrating the visualization and analysis software originally created by [IOBIO](https://iobio.io/).
```

**Comment by MitchellShiell** [comment]
Looks great happy to see the template in action, let me know if their's is anything you'd like to see that is not included, or should be removed, LGTM Dan!

File: README.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+- This project is part of the [Overture](https://www.overture.bio/) research software ecosystem. See our [related products](#related-products) for more information on how Overture is helping organize data and enable discovery.
+
+</br>
+
+> 
+> <div>
+> <img align="left" src="ov-logo.png" height="90"/>
+> </div>
+> 
+> *Overture Iobio Components is part of [Overture](https://www.overture.bio/), a collection of open-source software microservices used to create platforms for researchers to organize and share genomics data.*
```
[the remark this answers is not in the document]
Comment (MitchellShiell) [reply README.md:14]: If commit, adjust ov-logo size appropriately

File: README.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+- [PNPM](https://pnpm.io/) Project manager
+- [Node.js](https://nodejs.org/en) Runtime environment (v20 or higher)
+- [VS Code](https://code.visualstudio.com/) As recommended code editor. Plugins recommended: ESLint, Prettier - Code formatter, Mocha Test Explorer, Monorepo Workspace
+
+### Setup
+
+1. Install the dependencies
+
+- This project is a monorepo managed by [PNPM](https://pnpm.io)
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply README.md:49]: I'll move this line into 'Development tools'

**Comment by demariadaniel** [comment]
> Looks great happy to see the template in action, let me know if their's is anything you'd like to see that is not included, or should be removed, LGTM Dan!

Thanks @MitchellShiell ! I've incorporated your feedback

**Review by MitchellShiell — APPROVED** [review]

**Review by joneubank — APPROVED** [review]
